### PR TITLE
Forward port release changes from v1.5.1

### PR DESCRIPTION
**Pull request by pb8o**
## Changes

Forward port release changes from v1.5.1 back to `main`

In addition, 

- Adopt "Keep a Changelog"
- fix some tests to honor `--binary-dir`

## Reason

Make the next release simpler.

## License Acceptance

By submitting this pull request, I confirm that my contribution is made under
the terms of the Apache 2.0 license. For more information on following
Developer Certificate of Origin and signing off your commits, please check
[`CONTRIBUTING.md`][3].

## PR Checklist

- [ ] If a specific issue led to this PR, this PR closes the issue.
- [x] The description of changes is clear and encompassing.
- [x] Any required documentation changes (code and docs) are included in this PR.
- [ ] API changes follow the [Runbook for Firecracker API changes][2].
- [ ] User-facing changes are mentioned in `CHANGELOG.md`.
- [ ] All added/changed functionality is tested.
- [ ] New `TODO`s link to an issue.
- [x] Commits meet [contribution quality standards](https://github.com/firecracker-microvm/firecracker/blob/main/CONTRIBUTING.md#contribution-quality-standards).

---

- [x] This functionality cannot be added in [`rust-vmm`][1].

[1]: https://github.com/rust-vmm
[2]: https://github.com/firecracker-microvm/firecracker/blob/main/docs/api-change-runbook.md
[3]: https://github.com/firecracker-microvm/firecracker/blob/main/CONTRIBUTING.md

### CODE REVIEW

**Comment by roypat** [comment]
Something about this breaks the memory overhead test, it seems like all the tests get deselected somehow :thinking: https://buildkite.com/firecracker/performance-a-b-tests/builds/160

**Comment by pb8o** [comment]
> Something about this breaks the memory overhead test, it seems like all the tests get deselected somehow 🤔 https://buildkite.com/firecracker/performance-a-b-tests/builds/160

Seems like the `-m nonci` should not be there? that part of the rebase was a bit tricky.

**Comment by roypat** [comment]
> > Something about this breaks the memory overhead test, it seems like all the tests get deselected somehow 🤔 https://buildkite.com/firecracker/performance-a-b-tests/builds/160
> 
> Seems like the `-m nonci` should not be there? that part of the rebase was a bit tricky.

Ahh, yeah, it shouldnt be. It seems like it tries to do a non-A/B run of all the tests

**Review by roypat — DISMISSED** [review]

File: tools/release-notes.sh [hunk 1]
```diff
@@ -26,7 +26,7 @@ changelog="$FC_ROOT_DIR/CHANGELOG.md"
 
 # Patterns for the sections in the changelog corresponding to the versions.
 pat_curr="/^##\s\[$curr_ver\]/"
-pat_prev="/^##\s\[.+]/ && NR>3"
+pat_prev="/^##\s\[.+]/ && NR>8"
```
Comment (kalyazin) [inline tools/release-notes.sh:29]: that's rather cryptic, maybe worth mentioning in the commit message?

File: docs/jailer.md [hunk 2]
```diff
@@ -282,3 +282,8 @@ Note: default value for `<api-sock>` is `/run/firecracker.socket`.
 - If all the cgroup controllers are bunched up on a single mount point using
   the "all" option, our current program logic will complain it cannot detect
   individual controller mount points.
+
+- [#4287](https://github.com/firecracker-microvm/firecracker/issues/4287) When
+  starting a jailer with `--parent-cgroup` specified but no cgroup flags
+  specified, then the rules in the parent cgroup folder are ignored. To
+  workaround, use a dummy cgroup parameter like `--cgroup=memory.max=max`.
```
[the remark this answers is not in the document]
Comment (pb8o) [reply docs/jailer.md:289]: oh, so it cannot be used as a verb. Thanks! Fixed!

File: CHANGELOG.md [hunk 3]
```diff
@@ -35,6 +35,10 @@
   containing vhost-user block devices.
   See the [related doc page](./docs/api_requests/block-vhost-user.md) for details.
   The device emits metrics under the label `"vhost_user_{device}_{drive_id}"`.
+- [#4287](https://github.com/firecracker-microvm/firecracker/issues/4287)
```
[the remark this answers is not in the document]
Comment (pb8o) [reply CHANGELOG.md:43]: Hmm so in this case there is no PR (I could use this one or the original one in v1.5.1). Should I remove the link completely, or just link it inline?

[the remark this answers is not in the document]
Comment (kalyazin) [reply CHANGELOG.md:43]: no, I think _a link_ is better than no link at all. Inlining it doesn't look bad to me.

**Review by kalyazin — APPROVED** [review]

**Pull request merged**